File: app/src/main/java/com/cqebd/student/live/custom/CustomAttachment.java
package com.cqebd.student.live.custom;

import com.netease.nimlib.sdk.msg.attachment.MsgAttachment;

/**
 * Created by xiaofu on 2018/5/30.
 */
public abstract class CustomAttachment implements MsgAttachment {

    protected int type;

    CustomAttachment(int type) {
        this.type = type;
    }

    public void fromJson(String data) {
        if (data != null) {
            parseData(data);
        }
    }

    @Override
    public String toJson(boolean send) {
        return packData();
    }

    public int getType() {
        return type;
    }

    protected abstract void parseData(String data);

    protected abstract String packData();
}

File: lib_netease/src/main/java/com/cqebd/lib_netease/customMsg/CustomAttachment.kt
package com.cqebd.lib_netease.customMsg

import com.netease.nimlib.sdk.msg.attachment.MsgAttachment


abstract class CustomAttachment : MsgAttachment {

    // 自定义消息附件的类型，根据该字段区分不同的自定义消息
//    protected var type: Int = 0

    // 解析附件内容。
    fun fromJson(data: String) {
        parseData(data)
    }

    override fun toJson(send: Boolean): String {
        return packData()
    }

    // 子类的解析和封装接口。
    protected abstract fun parseData(data: String)

    protected abstract fun packData():String

}